package-testing/tests/validate_a_new_module.rb

```diff
@@ -19,6 +19,10 @@
       on(workstation, "test -f #{module_name}/Gemfile.lock", accept_all_exit_codes: true) do |lock_check_outcome|
         assert_equal(0, lock_check_outcome.exit_code, 'pdk validate should have caused a Gemfile.lock file to be created')
       end
+
+      on(workstation, "cmp #{install_dir(workstation)}/share/cache/Gemfile.lock #{module_name}/Gemfile.lock", accept_all_exit_codes: true) do |gemfile_cmp|
+        assert_equal(0, gemfile_cmp.exit_code, 'newly created Gemfile.lock should match vendored Gemfile.lock')
```

Comment:
@scotje added a commit for this, lemme know what you think and I can squash and merge